### PR TITLE
[QF-4287] remove profile picture modal

### PR DESCRIPTION
**Pull request by afifvdin**
## Summary

Added a confirmation modal for deleting profile pictures to prevent accidental deletions.

Closes: [QF-4287](https://quranfoundation.atlassian.net/browse/QF-4287)

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [ ] ♻️ Refactoring (no functional changes)

## Environment Variables

_No new environment variables introduced._

## Test Plan

- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [x] Manual testing performed

**Testing steps:**

1. Navigate to Profile > Personalization page
2. Upload a profile picture if none exists
3. Click "Remove Picture" button
4. Verify confirmation modal appears.
5. Click the "Yes" button - modal should close and profile picture should be removed
6. Verify the modal can be closed by clicking outside without deleting

### Code Quality

- [x] I have performed a **self-review** of my code (file by file)
- [x] My code follows the [project style guidelines](/.github/copilot-instructions.md)
- [x] No `any` types used (or justified if unavoidable)
- [x] No unused code, imports, or dead code included
- [x] Complex logic has inline comments explaining "why"
- [x] Functions are under 30 lines and follow single responsibility

### Testing & Validation

- [ ] All tests pass locally (`yarn test`)
- [x] Linting passes (`yarn lint`)
- [x] Build succeeds (`yarn build`)
- [ ] Edge cases and error scenarios are handled

### Documentation

- [ ] Code is self-documenting with clear naming
- [ ] README updated (if adding features or setup changes)
- [ ] Inline comments added for complex logic

### Localization (if UI changes)

- [x] All user-facing text uses `next-translate`
- [x] Only English locale files modified (Lokalise handles others)
- [x] RTL layout verified

### Accessibility (if UI changes)

- [ ] Semantic HTML elements used
- [ ] ARIA attributes added where needed
- [ ] Keyboard navigation works

## Screenshots/Videos

| Before | After |
| ------ | ----- |
||image|

## Related PRs

_No related PRs._

## AI Assistance Disclosure

- [x] AI tools were NOT used for this PR
- [ ] AI tools were used, and I have **thoroughly reviewed and validated** all generated code

[QF-4287]: https://quranfoundation.atlassian.net/browse/QF-4287?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/Profile/DeleteProfilePictureButton.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        variant={ButtonVariant.Compact}
+        size={ButtonSize.Small}
+        className={styles.profilePictureActionButton}
+        onClick={onDeleteProfilePictureClicked}
+        isLoading={isRemoving}
+        isDisabled={isProcessing || isRemoving}
+      >
+        {t('remove-picture')}
+      </Button>
+      <Modal isOpen={isModalVisible} onClickOutside={closeModal}>
```
[the remark this answers is not in the document]
Comment (afifvdin) [reply src/components/Profile/DeleteProfilePictureButton.tsx:53]: let it as is

File: src/components/Profile/DeleteProfilePictureButton.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                className={styles.actionButton}
+                onClick={onDeleteConfirmed}
+                isDisabled={isRemoving}
+              >
+                {t('delete-profile-picture.action-text')}
+              </Button>
+            </div>
+          </Modal.Footer>
+        </Modal.Body>
+      </Modal>
```
[the remark this answers is not in the document]
Comment (afifvdin) [reply src/components/Profile/DeleteProfilePictureButton.tsx:80]: we dont need any input field

File: locales/en/profile.json [hunk 3]
```diff
@@ -7,6 +7,12 @@
     "subtitle": "This action cannot be undone. This will permanently delete your account, saved bookmarks, reading sessions, and all your data.",
     "title": "Are you absolutely sure?"
   },
+  "delete-profile-picture": {
+    "action-text": "Yes",
+    "cancel-text": "No",
```
[the remark this answers is not in the document]
Comment (afifvdin) [reply locales/en/profile.json:12]: following the story

File: src/components/Profile/DeleteProfilePictureButton.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+  const closeModal = useCallback(() => {
+    setIsModalVisible(false);
+  }, []);
+
+  const onDeleteConfirmed = useCallback(async () => {
+    logButtonClick('profile_confirm_delete_profile_picture');
+    onRemovePicture();
+    closeModal();
+  }, [onRemovePicture, closeModal]);
```
[the remark this answers is not in the document]
Comment (afifvdin) [reply src/components/Profile/DeleteProfilePictureButton.tsx]: yeah, should remove the async

File: locales/en/profile.json [hunk 5]
```diff
@@ -7,6 +7,12 @@
...
+    "subtitle": "You will delete the profile pic and reset it to the default, Are you sure?",
```
[the remark this answers is not in the document]
Comment (afifvdin) [reply locales/en/profile.json:13]: let it as is

File: src/components/Profile/DeleteProfilePictureButton.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        isLoading={isRemoving}
+        isDisabled={isProcessing || isRemoving}
+      >
+        {t('remove-picture')}
+      </Button>
+      <Modal isOpen={isModalVisible} onClickOutside={closeModal}>
+        <Modal.Body>
+          <Modal.Header>
+            <Modal.Title>{t('delete-profile-picture.title')}</Modal.Title>
+            <Modal.Subtitle>{t('delete-profile-picture.subtitle')}</Modal.Subtitle>
```
[the remark this answers is not in the document]
Comment (afifvdin) [reply src/components/Profile/DeleteProfilePictureButton.tsx:57]: following the code pattern
